_studio/mfx_lib/pkg-config.pc.cmake

```diff
@@ -3,7 +3,8 @@ Description: Intel(R) Media SDK
 Version: @mfx_version_major@.@mfx_version_minor@
 
 prefix=@CMAKE_INSTALL_PREFIX@
-libdir=@CMAKE_INSTALL_FULL_LIBDIR@
-includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
+libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@
+includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
```

Comment:
I usually configure as follows:

`cmake -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_INSTALL_LIBDIR=/usr/lib64...`

Will your change produce:

`includedir=/usr//usr/lib64`

which is eventually wrong?

Reply:
You might need to have something like the following in cmake:

`string(REPLACE ${CMAKE_INSTALL_PREFIX} \${prefix} OUT_VAR ${CMAKE_INSTALL_FULL_LIBDIR})`

I am not sure I have put it in correct cmake language... just giving an idea...

Reply:
Yep it will. But given defaults listed at https://cmake.org/cmake/help/latest/command/install.html#command:install you do it wrong way. Correct way to specify that would be: `cmake -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_INSTALL_LIBDIR=lib64`

Reply:
Looks GNUInstallDirs module does exactly this internally.

Reply:
> Looks GNUInstallDirs module does exactly this internally.

`cmake -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_INSTALL_LIBDIR=/usr/lib64 ..`

You mean that even this will actually work?

Reply:
Let me try to explain my point. Line: `cmake -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_INSTALL_LIBDIR=/usr/lib64 ..` is malformed. Previously it did work, but from GNUInstallDirs module perspective CMAKE_INSTALL_LIBDIR_FULL derived from CMAKE_INSTALL_PREFIX and CMAKE_INSTALL_LIBDIR or specified explicitly, but not vice-versa. Also there are few special values like "/" in prefix handling.

Reply:
It is not malformed. Read https://cmake.org/cmake/help/v3.18/module/GNUInstallDirs.html carefully:

> CMAKE_INSTALL_FULL_<dir>

> The absolute path generated from the corresponding CMAKE_INSTALL_dir value. If the value is not already an absolute path, an absolute path is constructed typically by prepending the value of the CMAKE_INSTALL_PREFIX variable. However, there are some special cases as documented below.

In prev. documentation it was a little more clear from my perspective:

https://cmake.org/cmake/help/v3.0/module/GNUInstallDirs.html

> Each CMAKE_INSTALL_<dir> value may be passed to the DESTINATION options of install() commands for the corresponding file type. If the includer does not define a value the above-shown default will be used and the value will appear in the cache for editing by the user. Each CMAKE_INSTALL_FULL_<dir> value contains an absolute path constructed from the corresponding destination by prepending (if necessary) the value of CMAKE_INSTALL_PREFIX.